Framework/Core/include/Framework/AnalysisDataModel.h

```diff
@@ -265,6 +265,127 @@ using TrackExtra = TracksExtra::iterator;
+DECLARE_SOA_COLUMN(SigmaTgl, sigmaTgl, float);
+DECLARE_SOA_COLUMN(Sigma1Pt, sigma1Pt, float);
+DECLARE_SOA_COLUMN(RhoXY, rhoXY, int8_t);
+DECLARE_SOA_COLUMN(RhoPhiX, rhoPhiX, int8_t);
+DECLARE_SOA_COLUMN(RhoPhiY, rhoPhiY, int8_t);
+DECLARE_SOA_COLUMN(RhoTglX, rhoTglX, int8_t);
+DECLARE_SOA_COLUMN(RhoTglY, rhoTglY, int8_t);
+DECLARE_SOA_COLUMN(RhoTglPhi, rhoTglPhi, int8_t);
+DECLARE_SOA_COLUMN(Rho1PtX, rho1PtX, int8_t);
+DECLARE_SOA_COLUMN(Rho1PtY, rho1PtY, int8_t);
+DECLARE_SOA_COLUMN(Rho1PtPhi, rho1PtPhi, int8_t);
+DECLARE_SOA_COLUMN(Rho1PtTgl, rho1PtTgl, int8_t);
+
+DECLARE_SOA_EXPRESSION_COLUMN(CXX, cXX, float, aod::trackfwd::sigmaX* aod::trackfwd::sigmaX);
+DECLARE_SOA_EXPRESSION_COLUMN(CXY, cXY, float, (aod::trackfwd::rhoXY / 128.f) * (aod::trackfwd::sigmaX * aod::trackfwd::sigmaY)); //Why /128.f ?
+DECLARE_SOA_EXPRESSION_COLUMN(CYY, cYY, float, aod::trackfwd::sigmaY* aod::trackfwd::sigmaY);
+DECLARE_SOA_EXPRESSION_COLUMN(CPhiX, cPhiX, float, (aod::trackfwd::rhoPhiX / 128.f) * (aod::trackfwd::sigmaPhi * aod::trackfwd::sigmaX));
+DECLARE_SOA_EXPRESSION_COLUMN(CPhiY, cPhiY, float, (aod::trackfwd::rhoPhiY / 128.f) * (aod::trackfwd::sigmaPhi * aod::trackfwd::sigmaY));
+DECLARE_SOA_EXPRESSION_COLUMN(CPhiPhi, cPhiPhi, float, aod::trackfwd::sigmaPhi* aod::trackfwd::sigmaPhi);
+DECLARE_SOA_EXPRESSION_COLUMN(CTglX, cTglX, float, (aod::trackfwd::rhoTglX / 128.f) * (aod::trackfwd::sigmaTgl * aod::trackfwd::sigmaX));
+DECLARE_SOA_EXPRESSION_COLUMN(CTglY, cTglY, float, (aod::trackfwd::rhoTglY / 128.f) * (aod::trackfwd::sigmaTgl * aod::trackfwd::sigmaY));
+DECLARE_SOA_EXPRESSION_COLUMN(CTglPhi, cTglPhi, float, (aod::trackfwd::rhoTglPhi / 128.f) * (aod::trackfwd::sigmaTgl * aod::trackfwd::sigmaPhi));
+DECLARE_SOA_EXPRESSION_COLUMN(CTglTgl, cTglTgl, float, aod::trackfwd::sigmaTgl* aod::trackfwd::sigmaTgl);
+DECLARE_SOA_EXPRESSION_COLUMN(C1PtY, c1PtY, float, (aod::trackfwd::rho1PtY / 128.f) * (aod::trackfwd::sigma1Pt * aod::trackfwd::sigmaY));
+DECLARE_SOA_EXPRESSION_COLUMN(C1PtX, c1PtX, float, (aod::trackfwd::rho1PtX / 128.f) * (aod::trackfwd::sigma1Pt * aod::trackfwd::sigmaX));
+DECLARE_SOA_EXPRESSION_COLUMN(C1PtPhi, c1PtPhi, float, (aod::trackfwd::rho1PtPhi / 128.f) * (aod::trackfwd::sigma1Pt * aod::trackfwd::sigmaPhi));
+DECLARE_SOA_EXPRESSION_COLUMN(C1PtTgl, c1PtTgl, float, (aod::trackfwd::rho1PtTgl / 128.f) * (aod::trackfwd::sigma1Pt * aod::trackfwd::sigmaTgl));
+DECLARE_SOA_EXPRESSION_COLUMN(C1Pt21Pt2, c1Pt21Pt2, float, aod::trackfwd::sigma1Pt* aod::trackfwd::sigma1Pt);
+
+// TRACKFWDEXTRA TABLE definition
+DECLARE_SOA_COLUMN(Chi2, chi2, float);
+//DECLARE_SOA_COLUMN(PDca, pDca, float);
+//DECLARE_SOA_COLUMN(RAtAbsorberEnd, rAtAbsorberEnd, float);
+// Expression columns for PDca and RAtAbsorberEnd will be replaced by SOA columns
+DECLARE_SOA_EXPRESSION_COLUMN(PDca, pDca, float, (aod::trackfwd::p)*nsqrt((aod::trackfwd::x * aod::trackfwd::x) + (aod::trackfwd::y * aod::trackfwd::y) + (aod::trackfwd::z * aod::trackfwd::z)));
```

Comment:
Irrespective of PDCA discussion above: why z component is accounted here? For prompt tracks PDCA is a measure of effective mult.scattering kick in transverse direction (dcaX, dcaY), while dcaZ is amplified by ``1/tg(theta)``.
Also, do ``aod::trackfwd::x, y`` have meaning of abs. coordinates at vertex or indeed DCA to it?

Reply:
@shahor02 as far as I understood, this calculation was just to get something to play with while developing the code, but indeed this is not the way it is computed in reality. It is in fact impossible to compute it with the info stored in AOD.

In the case of muon tracks (without MFT), aod::trackfwd::x, y is equal to the vertex position the track has been extrapolated to, so it cannot be used to compute the DCA. Instead we use the track extrapolated without Branson correction, which is not stored in the AOD.

Reply:
@pillot OK, but for MFT tracks this computation makes sense, provided the Z is suppressed (and delta to vertex is used).

Reply:
So for MFT tracks, we could reduce what is stored and use the computation but not for those which went through the absorber?

Reply:
For MFTStandalone and GlobalMuonTracks there is no need to store pDCA, as it can be calculated on the fly. Storing pDCA from reconstruction is necessary only for run2-like MCH tracks, which will be renamed to MUONStandalone in `Framework/DataTypes.h` (due popular vote). 

At the moment we do not foresee a table for pDCA evaluated on-the fly for the MFT, but there are several analysis possibilities considered. I mean that we are learning how to use MFTStandalone tracks and GlobalMuonTracks; the optimal data structures are yet to be unveiled.